### PR TITLE
Add sharedValueFiles to all clusterGroup files in lieu of "magic" value files

### PR DESCRIPTION
**Pull request by mhjacks**

### CODE REVIEW

File: values-all-in-one.yaml [hunk 1]
```diff
@@ -248,6 +248,10 @@ clusterGroup:
       # timeout: 234
       # verbosity: "-v"
 
+  sharedValueFiles:
+    - '/cluster-overrides/values-{{ $.Values.global.clusterPlatform }}.yaml'
+    - '/cluster-overrides/values-{{ $.Values.global.clusterPlatform }}-{{ $.Values.global.clusterVersion }}.yaml'
```
Comment (beekhof) [inline values-all-in-one.yaml:253]: Maybe?

```suggestion
    - '/overrides/cluster/values-{{ $.Values.global.clusterPlatform }}.yaml'
    - '/overrides/cluster/values-{{ $.Values.global.clusterPlatform }}-{{ $.Values.global.clusterVersion }}.yaml'
```

File: values-development.yaml [hunk 2]
```diff
@@ -135,14 +135,18 @@ clusterGroup:
...
     dev-spring-petclinic:
       name: dev-spring-petclinic
       namespace: devsecops-dev
       project: spring-petclinic
       kustomize: true
       path: charts/region/spring-petclinic/environments/dev
 
+  sharedValueFiles:
+    - '/cluster-overrides/values-{{ $.Values.global.clusterPlatform }}.yaml'
+E   - '/cluster-overrides/values-{{ $.Values.global.clusterPlatform }}-{{ $.Values.global.clusterVersion }}.yaml' #   E:
```
Comment (beekhof) [inline values-development.yaml:148]: Typo?
```suggestion
   - '/cluster-overrides/values-{{ $.Values.global.clusterPlatform }}-{{ $.Values.global.clusterVersion }}.yaml' #   E:
```

Comment (mhjacks) [reply values-development.yaml]: Yes. An artifact from copy-paste in vim. Fixed.

File: values-all-in-one.yaml (hunk 1 above)
Comment (mbaldessari) [reply values-all-in-one.yaml:253]: I also don't have an obvious winning proposal. I would avoid two folders for simplicity sake. Right now all our overrides are somehow cluster related so /cluster-overrides is not wrong. Maybe let's just call it `/overrides` for the time being and reassess later on?

Comment (mhjacks) [reply values-all-in-one.yaml:253]: I think `/overrides` makes sense. The key thing is to get them out of `/`; like you say the "magic" variables are all cluster-wide. Let's do that.

Comment (beekhof) [reply values-all-in-one.yaml]: `/overrides/{appname}/...` for the random stuff?
or simply `/overrides/{appname}-{some global var}.yaml`

The best part of having them be explicit, is that ultimately the consumer can arrange them any way they like. What we're doing here is setting up the cargo-cult baseline.  Starting with `/overrides` seems fine for that

File: values-development.yaml [hunk 3]
```diff
@@ -135,14 +135,18 @@ clusterGroup:
...
       name: dev-spring-petclinic
       namespace: devsecops-dev
       project: spring-petclinic
       kustomize: true
       path: charts/region/spring-petclinic/environments/dev
 
+  sharedValueFiles:
+    - '/overrides/values-{{ $.Values.global.clusterPlatform }}.yaml'
+    - '/overrides/values-{{ $.Values.global.clusterPlatform }}-{{ $.Values.global.clusterVersion }}.yaml' #   E:
+
```
Comment (beekhof) [inline values-development.yaml:149]: Move this to just before the applications section?

**Review by claudiol — APPROVED** [review]
/lgtm Approved

**Pull request merged**